app/Jobs/RockTheVote/CreateRockTheVoteReport.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Chompy\Jobs;
+
+use Illuminate\Bus\Queueable;
+use Chompy\Models\RockTheVoteReport;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+
+class CreateRockTheVoteReport implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable;
+
+    /**
+     * The since parameter to create a Rock The Vote report with.
+     *
+     * @var DateTime
+     */
+    protected $since;
+
+    /**
+     * The before parameter to create a Rock The Vote report with.
+     *
+     * @var DateTime
+     */
+    protected $before;
+
+    /**
+     * The Rock The Vote report created upon success.
+     *
+     * @var DateTime
+     */
+    protected $report;
+
+    /**
+     * Create a new job instance.
+     *
+     * @param DateTime $since
+     * @param DateTime $before
+     * @return void
+     */
+    public function __construct($since, $before)
+    {
+        $this->since = $since;
+        $this->before = $before;
+    }
+
+    /**
+     * Execute the job to create a Rock The Vote report and import it after creation.
+     *
+     * @return array
+     */
+    public function handle()
+    {
+        info('Creating report', ['before' => $this->before, 'since' => $this->since]);
+
+        $this->report = RockTheVoteReport::createViaApi($this->since, $this->before);
+
+        ImportRockTheVoteReport::dispatch(null, $this->report);
```

Comment:
Everything looks fine! Just a question about line 60...why is null included on the dispatch?

Reply:
We're sending `null` for the `user` parameter because this job is being dispatched by the console command. We set it to the authenticated user when we create a report via the web over in the `RockTheVoteReportController`:
https://github.com/DoSomething/chompy/blob/2c8fa724eda9e8077b1c8f5c69e91a12869aa622/app/Http/Controllers/Web/RockTheVoteReportController.php#L50

Thank you for reviewing! 😸

Reply:
ohhh ok got it! And you're welcome @aaronschachter 🙂